src/registry/LocalRegistry.ts

```diff
@@ -89,6 +91,65 @@ export class LocalRegistry extends BaseRegistry implements IRegistry {
     return addresses[chainName];
   }
 
+  addChains(
+    chains: Array<{ chainName: ChainName; metadata?: ChainMetadata; addresses?: ChainAddresses }>,
+  ): void {
+    const currentChains = this.listRegistryContent().chains;
+    for (const chain of chains) {
+      if (currentChains[chain.chainName])
+        throw new Error(`Chain ${chain.chainName} already exists in registry`);
+    }
+
+    this.updateChains(chains);
+  }
+
+  updateChains(
+    chains: Array<{ chainName: ChainName; metadata?: ChainMetadata; addresses?: ChainAddresses }>,
+  ): void {
+    for (const chain of chains) {
+      if (!chain.metadata && !chain.addresses)
+        throw new Error(
+          `Chain ${chain.chainName} must have metadata or addresses, preferably both`,
+        );
+    }
+
+    // Ensure caches are initialized and filled before making changes
+    this.listRegistryContent();
+    this.getMetadata();
+    this.getAddresses();
+
+    for (const chain of chains) {
+      const chainName = chain.chainName;
+      if (chain.metadata) {
+        this.createChainFile(
+          chainName,
+          'metadata',
+          chain.metadata,
+          this.metadataCache!,
+          CHAIN_SCHEMA_REF,
+        );
+      }
+      if (chain.addresses) {
+        this.createChainFile(chainName, 'addresses', chain.addresses, this.addressCache!);
```

Comment:
would we ever add an addresses schema later?

Reply:
We can if you'd like, we just need a set list of contract names. Gonna punt on it for now